spec/controllers/openid_connect/authorization_controller_spec.rb

```diff
@@ -167,6 +173,151 @@
           end
         end
 
+        context 'with ialmax requested' do
+          before { params[:acr_values] = Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF }
+
+          context 'account is already verified' do
+            let(:user) do
+              create(
+                :profile, :active, :verified, proofing_components: { liveness_check: true }
+              ).user
+            end
+
+            it 'redirects to the redirect_uri immediately when pii is unlocked' do
+              IdentityLinker.new(user, service_provider).link_identity(ial: 3)
+              user.identities.last.update!(
+                verified_attributes: %w[given_name family_name birthdate verified_at],
+              )
+              allow(controller).to receive(:pii_requested_but_locked?).and_return(false)
+              action
+
+              expect(response).to redirect_to(/^#{params[:redirect_uri]}/)
+            end
+
+            it 'redirects to the password capture url when pii is locked' do
+              IdentityLinker.new(user, service_provider).link_identity(ial: 3)
+              user.identities.last.update!(
+                verified_attributes: %w[given_name family_name birthdate verified_at],
+              )
+              allow(controller).to receive(:pii_requested_but_locked?).and_return(true)
+              action
+
+              expect(response).to redirect_to(capture_password_url)
+            end
+
+            it 'tracks IAL2 authentication event' do
+              stub_analytics
+              expect(@analytics).to receive(:track_event).
+                with(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                     success: true,
+                     client_id: client_id,
+                     errors: {},
+                     unauthorized_scope: false,
+                     user_fully_authenticated: true,
+                     acr_values: 'http://idmanagement.gov/ns/assurance/ial/0',
+                     scope: 'openid profile')
+              expect(@analytics).to receive(:track_event).
+                with(
+                  Analytics::SP_REDIRECT_INITIATED,
+                  ial: 0,
+                  billed_ial: 2,
+                )
+
+              IdentityLinker.new(user, service_provider).link_identity(ial: 2)
+              user.identities.last.update!(
+                verified_attributes: %w[given_name family_name birthdate verified_at],
```

Comment:
note that these should not all be required but there's a bug with OIDC and IALMAX: https://cm-jira.usa.gov/browse/LG-6355